src/compiler/checker.ts

```diff
@@ -6646,7 +6647,7 @@ namespace ts {
             }
 
             function hasExcessProperties(source: FreshObjectLiteralType, target: Type, reportErrors: boolean): boolean {
-                if (!(target.flags & TypeFlags.ObjectLiteralPatternWithComputedProperties) && maybeTypeOfKind(target, TypeFlags.ObjectType)) {
+                if (maybeTypeOfKind(target, TypeFlags.ObjectType) && !(target as ObjectType).inObjectLiteralPatternWithComputedProperties) {

```

Comment:
The `maybeTypeOfKind` function could return true when `target` is a union or intersection containing an object type, so the second check needs to first ensure `target` is actually an object type.